src/Chapter2.hs

```diff
@@ -348,9 +361,13 @@ Implement a function that returns only the first half of a given list.
 >>> firstHalf "bca"
 "b"
 -}
--- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
-
+firstHalf :: [a] -> [a]
+firstHalf l
+  | null l = l
+  | otherwise = subList 0 (halfIndex (length l)) l
```

Comment:
Great! You reused `subList` in here 😮  👏🏼 

Alternatively, `sublist 0` is just the `take` function 🙂